docusaurus/docs/Android/04-compose/05-message-components/01-messages-screen.mdx

```diff
@@ -70,31 +75,66 @@ Given that the `MessagesScreen` is a screen level solution, it offers limited cu
 ```kotlin
 @Composable
 fun MessagesScreen(
-    messageLimit: Int = 30,
+    viewModelFactory: MessagesViewModelFactory,
     showHeader: Boolean = true,
-    enforceUniqueReactions: Boolean = true,
-    showDateSeparators: Boolean = true,
-    showSystemMessages: Boolean = true,
-    deletedMessageVisibility: DeletedMessageVisibility = DeletedMessageVisibility.ALWAYS_VISIBLE,
-    messageFooterVisibility: MessageFooterVisibility = MessageFooterVisibility.WithTimeDifference(),
     ... // action handlers and state
 )
 ```
 
-* `messageLimit`: The number of messages per query.
+* `viewModelFactory`: The factory that you build yourself. This lets you control not just the way the `ViewModel`s are built, but also their lifecycle, as you can share them between components. This requires of you to provide a `channelId` in order to power the screen and show data, but it also allows you to customize the behavior of the screen through various parameters.
 * `showHeader`: Controls whether the messages header is shown or not.
-* `enforceUniqueReactions`: Controls whether the user can leave multiple reactions to a message.
-* `showDateSeparators`: Controls if we show date separator items.
-* `showSystemMessages`: Controls if we show system message items.
-* `deletedMessageVisibility`: Controls the visibility of deleted messages. By default, all deleted messages are visible in the "deleted" state.
-* `messageFooterVisibility`: Controls the visibility of the message footer. By default, determined by a time difference.
 
 If you set `showHeader` to `false` you'll get the following UI:
 
 ![MessagesScreen without the MessageListHeader](../../assets/compose_custom_messages_screen_component.png)
 
 As you can see, the header is removed from the screen, rendering only the list and the composer.
 
+## Overriding the ViewModels
+
+In case you want to control the logic when using the `MessagesScreen`, you can do so by providing a `MessagesViewModelFactory` that you use to build the respective ViewModels yourself.
+
+Here's an example:
+
+```kotlin
+class MessagesActivity : BaseConnectedActivity() {
+
+    // 1
+    private val factory by lazy {
+        MessagesViewModelFactory(
+            context = this,
+            channelId = intent.getStringExtra(KEY_CHANNEL_ID) ?: "",
```

Comment:
Does it make sense to provide an empty cid? Maybe we should suggest `requireNotNull()` instead? 🤔

Reply:
Yeah definitely makes sense!